services/web/client/source/class/qxapp/component/filter/TagsFilter.js

```diff
@@ -0,0 +1,126 @@
+/* ************************************************************************
+
+   qxapp - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2019 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Ignacio Pascual (ignapas)
+
+************************************************************************ */
+
+/**
+ * Dropdown menu for tag filtering
+ */
+
+qx.Class.define("qxapp.component.filter.TagsFilter", {
+  extend: qxapp.component.filter.UIFilter,
+
+  construct: function(filterId, groupId, labelTr = "Tags") {
+    this.base(arguments, filterId, groupId);
+    this._setLayout(new qx.ui.layout.HBox());
+
+    this.__dropDown = new qx.ui.toolbar.MenuButton(this.tr(labelTr));
+    this.__dropDown.setMenu(this.__buildMenu());
+    this._add(this.__dropDown);
+  },
+
+  members: {
+    __dropDown: null,
+    __activeTags: null,
+    __tagButtons: null,
+
+    reset: function() {
+      // Remove ticks from menu
+      const menuButtons = this.__dropDown.getMenu().getChildren()
+        .filter(child => child instanceof qx.ui.menu.Button);
+      menuButtons.forEach(button => button.resetIcon());
+      // Remove active tags
+      if (this.__activeTags && this.__activeTags.length) {
+        this.__activeTags.length = 0;
+      }
+      // Remove tag buttons
+      for (let tagName in this.__tagButtons) {
+        this._remove(this.__tagButtons[tagName]);
+        delete this.__tagButtons[tagName];
+      }
+      // Dispatch
+      this._filterChange(this.__activeTags);
+    },
+
+    __buildMenu: function() {
+      const menu = new qx.ui.menu.Menu();
+
+      this.__getServiceTypes().forEach(serviceType => {
+        const button = new qx.ui.menu.Button(qxapp.utils.Utils.capitalize(serviceType));
+        button.addListener("tap", e => this.__addTag(serviceType, e.getTarget()));
```

Comment:
why not listen to the 'execute' event? execute can also be issued from the keyboard whereas 'tap' is restricted to pointing devices.